Lesson 4/src/app/components/WelcomeModal.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faTimes} from '@fortawesome/free-solid-svg-icons';
+
+export default class WelcomeModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            modal: false
+        };
+        console.log('1. constructor()');
+    }
+
+    static getDerivedStateFromProps(props, state){
+        console.log('2. getDerivedStateFromProps(props, state)');
+        return null
+    }
+
+    render() {
+        console.log('3. render()');
+        let modal;
+        if (this.state.modal){
+            modal =
+                <>
+                    <div className={'WelcomeModal'}>
+                        <div className={'container trans slideDown'}>
+                            <div className={'titleWrap'}>
+                                <h3 className={'title'}>{this.props.items.title}</h3>
+                            </div>
+                            <div className={'detailsWrap'}>
+                                <p className={'details'}>{this.props.items.modalBody}</p>
+                            </div>
+                            <div className={'btnWrap'}>
+                                <button className={'closeXbtn'}
+                                        onClick={()=>this.setState({modal:!this.state.modal})}
```

Comment:
```suggestion
                                        onClick={()=>this.setState({modal:!this.state.modal})}
```
лучше вынести данную функцию за пределы `render()`, так как при каждой перерисовке будет создаваться новая функция